Lind-Project/lind_project
# Test files for mknod/fchdir/poll

## Description
This PR introduces three new test cases for the mknod, fchdir, and poll syscalls. The test files have been created and tested for validity. They will help us confirm the correct operation of these syscalls.

### fchdir.c
A new C source file, `fchdir.c`, tests the `fchdir` system call functionality. This file opens a directory, changes the current working directory to the newly opened directory, and then verifies this operation by getting the current working directory again.

### mknod.c
We've introduced a C source file, `mknod.c`, to verify the `mknod` system call. This test creates and then removes a character device.

### poll.c
Lastly, a C source file, `poll.c`, has been added to evaluate the `poll` system call. This test script observes if it succeeds using the poll syscall.

### Type of change

## How Has This Been Tested?
To reproduce the tests, follow these steps for each system call:

### Test of mknod
- Test of `mknod`- `lind_project/tests/test_cases/mknod.c`
1. Compile `mknod.c` in `lindsh` with `compile tests/test_cases/mknod.c -o mknod.nexe`
2. Copy test file to Lind file system with `lindfs cp /home/lind/lind_project/mknod.nexe mknod.nexe`
3. If all operations were successful, your file should have been copied. You can check the file with `lindfs ls /` command.
4. Run the test file with `lind /mknod.nexe`

### Test of fchdir
- Test of `fchdir` - `lind_project/tests/test_cases/fchdir.c`
1. Compile `fchdir.c` in `lindsh` with `compile tests/test_cases/fchdir.c -o fchdir.nexe`
2. Copy test file to Lind file system with `lindfs cp /home/lind/lind_project/fchdir.nexe fchdir.nexe`
3. To perform the test, create a directory with the `lindfs mkdir automated_tests` command.
4. If all operations are successful, your file should be copied and your directory should be created. You can check the file with `lindfs ls /` command.
5. Run the test file with `lind /fchdir.nexe`

### Test of poll
- Test of `poll` - `lind_project/tests/test_cases/poll.c`
1. Compile `poll.c` in `lindsh` with `compile tests/test_cases/poll.c -o poll.nexe`
2. Copy test file to Lind file system with `lindfs cp /home/lind/lind_project/poll.nexe poll.nexe`
3. If all operations were successful, your file should have been copied. You can check the file with `lindfs ls /` command.
4. Run the test file with `lind /poll.nexe`

## Checklist:



- [x] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, safeposix-rust)
